### PR TITLE
Publish web pages to Skynet 

### PR DESCRIPTION
**Pull request by benton-fried36**
Added a new feature where users can create a branch called dg-pages and dgit will automatically push any frontend files to Skynet

### CODE REVIEW

**Comment by bqv** [comment]
I think this project is dead. You may want to fork

**Comment by tobowers** [comment]
@bqv 

Image of not dead parrothttps://media.giphy.com/media/paRbapEqsVqyA/giphy.gif)

The project isn't quite dead (but we've not been maintaining it as much as we'd like). We supported @littlezigy on this PR :).

File: initializer/initializer.go [hunk 1]
```diff
@@ -111,6 +112,73 @@ func (i *Initializer) Init(ctx context.Context, args []string) error {
...
+        touch _pages/_e2kdie_
+        skynet upload _pages
+        rm -rf _pages
+    fi
+done
+
+exit 0
+    `)
+
+    c:= []byte(`package main
```
Comment (tobowers) [inline initializer/initializer.go:147]: This part is probably best done as part of the CLI of dgit itself

**Comment by tobowers** [comment]
@littlezigy thank you for your contribution!

File: initializer/initializer.go (hunk 1 above)
Comment (littlezigy) [reply initializer/initializer.go:147]: Okk. Is there a manual for the commands? dgit --help doesn't bring up a lot of options

**Comment by tobowers** [comment]
@littlezigy unfortunately no :( but you can see how the other commands are implemented here: https://github.com/quorumcontrol/dgit/tree/01524ac6c0c16372f0d6a5a64da27544c55b2c4f/cmd

**Comment by c-nv-s** [comment]
great pull request idea! really hope this project will see some renewed love in the near future

File: initializer/initializer.go (hunk 1 above)
Comment (littlezigy) [reply initializer/initializer.go]: Sorry for awol, I've had a hectic two weeks. I moved the git hook creation block to init.go

File: README.md [hunk 2]
```diff
@@ -53,6 +53,10 @@ As an example:
 If you want to keep your decentralized, shareable git remote in sync with your GitHub repo adding
 a simple github action as illustrated in [dgit-github-action](https://github.com/quorumcontrol/dgit-github-action) is all it takes.  Once completed your decentragit decentralized shareable remote will always be up to date and ready when you need it.<br>
 
+#### Publish to dg-pages
+New Feature! DGit now allows you to publish your frontend files (html, css, js, .vue, .react) files to Skynet.
+Just checkout your files into a new branch named dg-pages, commit and push!
+
```
Comment (brandonwestcott) [inline README.md:59]: given the githook uses a the `skynet` executable, it should be noted in this section as a requirement.

File: cmd/init.go [hunk 3]
```diff
@@ -38,6 +39,45 @@ var initCommand = &cobra.Command{
...
+            remote=$1
+            remote_url="$2"
+
+
+            echo "Remote name is $remote"
+            echo "Remote URL is $remote_url"
+
+            while read local_ref remote_ref
+            do
+                echo "Local branch $local_ref"
```
Comment (brandonwestcott) [inline cmd/init.go:57]: Given this hook executes on every push, I would suggest removing these trace/debug lines from 46, 52-43, and 57. Optionally you can wrap them in a `"$DGIT_LOG_LEVEL" == "debug"` conditional

Comment (littlezigy) [reply cmd/init.go]: I totally forgot about those. Removed all echoes

**Review by brandonwestcott — APPROVED** [review]
looks great 👍 , thanks for the contribution!

**Comment by tobowers** [comment]
Thank you for your contribution!

**Pull request merged**